packages/edge-config/src/utils/tracing.ts

```diff
@@ -1,14 +1,82 @@
-import type { TracerProvider, Tracer } from '@opentelemetry/api';
+import type { TracerProvider, Tracer, Attributes } from '@opentelemetry/api';
 import { name as pkgName, version } from '../../package.json';
 
-let tracer: Tracer | undefined;
+let tracer: TracerProvider | undefined;
 
 export function setTracerProvider(tracerProvider: TracerProvider): void {
-  tracer = tracerProvider.getTracer(pkgName, version);
+  tracer = tracerProvider;
```

Comment:
Could you explain this approach? Normally, the tracer is resolved from the `@opentelementy/api`'s `trace.getTracer()`. Are you attempting to keep this strictly type dependency?

Reply:
Re-used what @dferber90 started, but the thought process was that someone who wants to trace the SDK can call `setTracerProvider` with whatever provider they have, potentially just `trace` from `@opentelemetry/api`.

Would it be better to not have `setTracerProvider` and just try to get `trace` from `@opentelemetry/api`?

Reply:
I'm not too familiar with otel, so using this approach was just based on how I understood the docs

https://opentelemetry.io/docs/concepts/instrumentation/libraries/#getting-a-tracer

<img width="822" alt="image" src="https://github.com/vercel/storage/assets/1765075/f42ffeb2-e096-4835-aca8-f403768affc8">

Totally happy to accept any guidance here